backend/src/main/kotlin/app/ehrenamtskarte/backend/freinet/webservice/FreinetApplicationMutationService.kt

```diff
@@ -44,37 +46,28 @@ class FreinetApplicationMutationService {
             }
 
             val freinetAgency = FreinetAgencyRepository.getFreinetAgencyByRegionId(regionId)
+                // Freinet is enabled for this region if dataTransferActivated is true
+                ?.takeIf { it.dataTransferActivated }
                 ?: return@transaction false
 
-            if (!freinetAgency.dataTransferActivated) {
-                // Freinet is disabled for this region
-                return@transaction false
-            }
-
-            val applicationData = application.jsonValue
             val objectMapper = jacksonObjectMapper()
-            val jsonNode = objectMapper.readTree(applicationData)
-
+            val jsonNode = objectMapper.readTree(application.jsonValue)
             val personalDataNode = jsonNode
                 .path("value").firstOrNull { it["name"].asText() == "personalData" }
 
-            fun JsonNode.findValueByName(fieldName: String): String? =
-                this.firstOrNull { it["name"].asText() == fieldName }
-                    ?.get("value")
-                    ?.asText()
-
             val firstName = personalDataNode?.get("value")?.findValueByName("forenames").orEmpty()
             val lastName = personalDataNode?.get("value")?.findValueByName("surname").orEmpty()
             val dateOfBirth = personalDataNode?.get("value")?.findValueByName("dateOfBirth").orEmpty()
 
-            val persons = FreinetSearchPersonApi.searchPerson(
-                host = projectConfig.freinet.host,
-                firstName = firstName,
-                lastName = lastName,
-                dateOfBirth = dateOfBirth,
-                accessKey = freinetAgency.apiAccessKey,
-                agencyId = freinetAgency.agencyId,
-            )
+            val persons = runBlocking {
```

Comment:
@andrew8er i guess if we use `runBlocking` here it will not solve the issue to unblock the api call
I would suggestion to wrap the api call with
```
CoroutineScope(Dispatchers.IO).launch {
// api call
}
```
Even it would be great to get some clarification why this makes sense and what happens behind the scenes